sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala

```diff
@@ -550,21 +564,37 @@ private[hive] class SparkSQLCLIDriver extends CliDriver with Logging {
           // Sample query: select "quoted value --"
           //                                    ^^ avoids starting a comment if it's inside quotes.
         } else if (hasNext && line.charAt(index + 1) == '-') {
-          // ignore quotes and ;
-          insideComment = true
+          // ignore quotes and ; in dash-comment
+          insideDashComment = true
         }
       } else if (line.charAt(index) == ';') {
         if (insideSingleQuote || insideDoubleQuote || insideComment) {
           // do not split
         } else {
-          // split, do not include ; itself
-          ret.add(line.substring(beginIndex, index))
+          if (contentBegin) {
+            // split, do not include ; itself
+            ret.add(line.substring(beginIndex, index))
+          }
           beginIndex = index + 1
+          contentBegin = false
         }
       } else if (line.charAt(index) == '\n') {
-        // with a new line the inline comment should end.
+        // with a new line the inline dash-comment should end.
         if (!escape) {
-          insideComment = false
+          insideDashComment = false
+        }
+      } else if (line.charAt(index) == '/' && !insideComment) {
+        val hasNext = index + 1 < line.length
+        if (insideSingleQuote || insideDoubleQuote) {
+          // Ignores '/' in any case of quotes
+        } else if (hasNext && line.charAt(index + 1) == '*') {
+          // ignore quotes and ; in bracketed comment
+          insideBracketedComment = true
+        }
+      } else if (line.charAt(index) == '/' && insideBracketedComment) {
+        if (line.charAt(index - 1) == '*') {
+          // record the right bound of bracketed comment
+          bracketedCommentRightBound = index
```

Comment:
We need a variable `bracketedCommentRightBound` for tracking a bracketed comment?  I think we can do it without it.